### PR TITLE
Refactor `node-exporter` Helm chart into component

### PR DESCRIPTION
**Pull request by shafeeqes**
**How to categorize this PR?**

/area dev-productivity quality robustness
/kind technical-debt test
/platform all

**What this PR does / why we need it**:

This PR refactors the `node-exporter` and adopts the component concept. Consequently, it eliminates the Helm chart and adds full unit tests, instead. This improves several aspects and the overall code quality.

**Which issue(s) this PR fixes**:
Part of #2754

**Special notes for your reviewer**:

**Release note**:

```other operator
NONE
```

### CODE REVIEW

**Comment by ary1992** [comment]
Looks good overall. Haven't tested it yet.

File: pkg/component/nodeexporter/nodexporter_suite_test.go [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+import (
+	"testing"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+func TestNodeexporter(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Botanist Component Node Exporter Suite")
```
Comment (acumino) [inline pkg/component/nodeexporter/nodexporter_suite_test.go]:
```suggestion
	RunSpecs(t, "Botanist Component NodeExporter Suite")
```

File: pkg/gardenlet/controller/shoot/shoot/reconciler_reconcile.go [hunk 2]
```diff
@@ -581,6 +581,13 @@ func (r *Reconciler) runReconcileShootFlow(ctx context.Context, o *operation.Ope
 			Fn:           flow.TaskFn(botanist.DeployAPIServerProxy).RetryUntilTimeout(defaultInterval, defaultTimeout).SkipIf(o.Shoot.IsWorkerless),
 			Dependencies: flow.NewTaskIDs(waitUntilGardenerResourceManagerReady, initializeShootClients, ensureShootClusterIdentity, deployKubeScheduler, waitUntilShootNamespacesReady),
 		})
+		_ = g.Add(flow.Task{
+			Name: "Deploying node-exporter",
+			Fn: flow.TaskFn(func(ctx context.Context) error {
+				return botanist.Shoot.Components.SystemComponents.NodeExporter.Deploy(ctx)
```
Comment (acumino) [inline pkg/gardenlet/controller/shoot/shoot/reconciler_reconcile.go]: We don't deploy node-exporter for every shoot, It is deployed only if shoot monitoring is enabled.

File: pkg/component/nodeexporter/nodexporter_suite_test.go [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+package nodeexporter_test
+
+import (
+	"testing"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+func TestNodeexporter(t *testing.T) {
```
Comment (acumino) [inline pkg/component/nodeexporter/nodexporter_suite_test.go]:
```suggestion
func TestNodeExporter(t *testing.T) {
```

File: charts/shoot-core/components/charts/monitoring/charts/node-exporter/templates/vpa.yaml [hunk 4]
```diff
@@ -1,20 +1,10 @@
+#TODO(shafeeqes): Remove this chart once gardener v1.75 is released
```
Comment (acumino) [inline charts/shoot-core/components/charts/monitoring/charts/node-exporter/templates/vpa.yaml]: Same change at other places.
```suggestion
# TODO(shafeeqes): Remove this chart once gardener v1.75 is released
```

File: pkg/operation/botanist/nodeexporter_test.go [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+
+var _ = Describe("NodeExporter", func() {
+	var (
+		ctrl     *gomock.Controller
+		botanist *Botanist
+	)
+
+	BeforeEach(func() {
+		ctrl = gomock.NewController(GinkgoT())
+		botanist = &Botanist{Operation: &operation.Operation{}}
```
Comment (acumino) [inline pkg/operation/botanist/nodeexporter_test.go]: This line is not needed.
```suggestion
```

File: pkg/operation/botanist/nodeexporter_test.go [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+		botanist.Shoot = &shootpkg.Shoot{}
+		botanist.Shoot.SetInfo(&gardencorev1beta1.Shoot{
+			Spec: gardencorev1beta1.ShootSpec{
+				Kubernetes: gardencorev1beta1.Kubernetes{
+					Version: "1.22.1",
+				},
+			},
+		})
+		botanist.Garden = &garden.Garden{}
```
Comment (acumino) [inline pkg/operation/botanist/nodeexporter_test.go]:
```suggestion
```

File: pkg/component/nodeexporter/nodeexporter.go [hunk 7]
```diff
@@ -0,0 +1,426 @@
...
+	// labelValue is the value of a label used for the identification of node-exporter pods.
+	labelValue = "node-exporter"
+
+	labelKeyComponent = "component"
+	serviceName       = "node-exporter"
+	daemonsetName     = "node-exporter"
+	containerName     = "node-exporter"
+
+	portNameMetrics           = "metrics"
+	portMetrics         int32 = 16909
```
Comment (acumino) [inline pkg/component/nodeexporter/nodeexporter.go]: This looks better.
```suggestion
	portMetrics         = int32(1234)
```

**Review by acumino — COMMENTED** [review]
Left few comments, Other than that you can move the `node-exporter` specific logging in `nodeexporter` package https://github.com/gardener/gardener/blob/809fd994e01455eb9f67af35f195bab76ebade2f/pkg/component/monitoring/logging.go#L44-L61

**Review by acumino — COMMENTED** [review]
/retest-required

File: pkg/component/nodeexporter/logging_test.go [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+								},
+							},
+						},
+					},
+				}))
+			Expect(loggingConfig.Parsers).To(Equal(
+				[]*fluentbitv1alpha2.ClusterParser{
+					{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:   "node-exporter" + "-parser",
```
Comment (acumino) [inline pkg/component/nodeexporter/logging_test.go]: Nit
```suggestion
							Name:   "node-exporter-parser",
```

File: pkg/component/nodeexporter/nodeexporter_test.go [hunk 9]
```diff
@@ -0,0 +1,481 @@
...
+	"github.com/gardener/gardener/pkg/client/kubernetes"
+	"github.com/gardener/gardener/pkg/component"
+	. "github.com/gardener/gardener/pkg/component/nodeexporter"
+	"github.com/gardener/gardener/pkg/utils/retry"
+	retryfake "github.com/gardener/gardener/pkg/utils/retry/fake"
+	"github.com/gardener/gardener/pkg/utils/test"
+	. "github.com/gardener/gardener/pkg/utils/test/matchers"
+)
+
+var _ = Describe("Nodeexporter", func() {
```
Comment (acumino) [inline pkg/component/nodeexporter/nodeexporter_test.go]:
```suggestion
var _ = Describe("NodeExporter", func() {
```

**Review by acumino — COMMENTED** [review]
/lgtm

**Review by acumino — COMMENTED** [review]
/approve

**Pull request merged**